dt-metrics/records/dynamic-records-map.php

```diff
@@ -252,9 +265,33 @@ public function points_geojson( WP_REST_Request $request ) {
         $query = ( isset( $params['query'] ) && !empty( $params['query'] ) ) ? $params['query'] : [];
         $query = dt_array_merge_recursive_distinct( $query, $this->base_filter );
 
-        return Disciple_Tools_Mapping_Queries::points_geojson( $post_type, $query );
+        return Disciple_Tools_Mapping_Queries::points_geojson( $post_type, self::capture_list_all_setting( $query ) );
     }
 
+    private function capture_list_all_setting( $params ) {
+        if ( !isset( $params['list_all'] ) || !$params['list_all'] ) {
+            $params['list_all'] = false;
+            $params['assigned_post_ids'] = self::list_assigned_post_ids( ( ( $params['post_type'] === 'system-users' ) ? 'contacts' : $params['post_type'] ), get_current_user_id() );
+        } else {
+            $params['list_all'] = true;
+            $params['assigned_post_ids'] = [];
+        }
+
+        return $params;
+    }
+
+    private function list_assigned_post_ids( $post_type, $user_id, $fields = [ 'ID' ], $limit = 500000 ): array {
```

Comment:
@kodinkat,  lets add an extra param post_type_geojson with the restrict param and add the sql directly if it is present.
Only query subassigned of the post_type is contacts.
Also remember that "subassinged" does not use the $user_id, it needs to be the contact id.